src/halmos/sevm.py

```diff
@@ -3106,15 +3122,9 @@ def finalize(ex: Exec):
                     src_offset = ex.int_of(ex.st.pop(), "symbolic MCOPY srcOffset")
                     size = ex.int_of(ex.st.pop(), "symbolic MCOPY size")
 
-                    if size > 0:
-                        src_end_loc = src_offset + size
-                        dst_end_loc = dest_offset + size
-
-                        if max(src_end_loc, dst_end_loc) > MAX_MEMORY_SIZE:
-                            raise HalmosException("MCOPY > MAX_MEMORY_SIZE")
-
-                        data = ex.st.memory.slice(src_offset, src_end_loc)
-                        ex.st.memory.set_slice(dest_offset, dst_end_loc, data)
+                    if size:
+                        data = ex.st.mslice(src_offset, size)
+                        ex.st.memory.set_slice(dest_offset, dest_offset + size, data)
```

Comment:
previously, both src and dst were checked, but now only src is checked. is it intentional, or is it implicitly checked during set_slice?

Reply:
you're right that's an oversight, I was focused on the case where size is huge, but dest_offset could also be huge

Reply:
dst is now checked in set_mslice